quanto/library/python/udqmm.py

```diff
@@ -0,0 +1,20 @@
+import torch
+
+from quanto.tensor.core import ungroup
+
+
+@torch.library.impl("quanto_py::udqmm", "default")
+def udqmm(
+    input: torch.Tensor,
+    weights: torch.Tensor,
+    scale: torch.Tensor,
+    zeropoint: torch.Tensor,
+    axis: int,
+    bits: int,
+    orig_shape: torch.Size,
+):
+    unpacked_weights = torch.ops.quanto.unpack(weights, bits)
+    shifted_weights = unpacked_weights.to(torch.int8) - zeropoint.to(torch.int8)
+    scaled_weights = shifted_weights * scale
```

Comment:
FYI, this will actually promote shifted_weigths to float16 then do the multiplication.
Since I had issues with these kind of implicit operation, maybe we could do that explicitly:
```
scaled_weights = shifted_weights.to(scale.dtype) * scale
```

Reply:
Good to know !